### PR TITLE
Update linter.py to remove deprecated SublimeText variables

### PR DESCRIPTION
**Pull request by robertsonwang**
Modified linter.py to handle the latest version of SublimeText. Breaking on installation otherwise.

### CODE REVIEW

File: linter.py [hunk 1]
```diff
@@ -40,19 +30,15 @@ class Lintr(Linter):
...
-    comment_re = r'#'
     tempfile_suffix = 'lintr'
 
     def cmd(self):
         """Return a list with the command line to execute."""
-        settings = self.get_view_settings()
+        settings = self.settings
         command = 'library(lintr);lint(cache = {0}, commandArgs(TRUE), {1})'.format(settings['cache'],
                                                                                     settings['linters'])
-        return [self.executable_path,
```
Comment (jimhester) [inline linter.py:55]: So does this mean R must be on the system `PATH`? I don't believe that was the case previously.

File: linter.py [hunk 2]
```diff
@@ -16,20 +16,10 @@
...
-        '-e',
-        'packageVersion(\'lintr\')'
-    ]
-    version_re = r'(?P<version>\d+\.\d+\.\d+)'
-    version_requirement = '>= 0.1.0'
     defaults = {
         'linters': 'default_linters',
-        'cache': 'TRUE'
+        'cache': 'TRUE',
+        'selector': 'source.r'
```
Comment (jimhester) [inline linter.py:22]: The convention is for R files to always have an uppercase R, is there a reason you changed this?

File: linter.py (hunk 1 above)
Comment (robertsonwang) [reply linter.py:55]: Unfortunately yes, I'm not aware of any other solution

File: linter.py (hunk 2 above)
Comment (robertsonwang) [reply linter.py:22]: From the linter settings documentation:

http://www.sublimelinter.com/en/stable/linter_settings.html#selector

> To find out what selector to use for given file type, use the “Tools > Developer > Show Scope Name” menu entry.

Doing this in an R file indicated that I should use source.r

Comment (jimhester) [reply linter.py:22]: Ok!

File: linter.py (hunk 1 above)
Comment (jimhester) [reply linter.py:55]: It looks like the python plugin has a `python` option to set where python is located. (http://www.sublimelinter.com/en/stable/linter_settings.html#python) I think we should probably add an `r` option so you can do the same thing.

Comment (robertsonwang) [reply linter.py:55]: Good idea! Will look into it

**Review by sbranders — APPROVED** [review]
I implemented those changes and added the R system path in the sublime lintr user settings.
`"paths":{"osx":["usr/local/bin/"]}`
It's working fine for me now.

**Comment by jimhester** [comment]
@robertsonwang do you think you could update the python version in `.travis.yml` so the tests will work? I think just bumping the version would suffice.

**Comment by robertsonwang** [comment]
sorry I haven't had time to look at this! Will bump the version in `.travis.yml` now

**Comment by robertsonwang** [comment]
Just merged in changes to my branch from @infotroph, everything passes :D

**Comment by jimhester** [comment]
Thanks!

**Pull request merged**